### ISSUE TITLE
Segfault in Envoy

### ISSUE THREAD

**Issue by kyessenov** [istio/istio#14707]
Reproduced twice with the following set-up:
- deploy bookinfo.yaml injected with 1.2.0-RC0 to namespace bookinfo
- `kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml -n bookinfo`
- rename productpage port name to "tcp" from "http"
- make sure bookinfo works
- `kubectl apply -f samples/bookinfo/policy/mixer-rule-deny-ip.yaml -n bookinfo`
- check productpage istio-proxy log for the crash

```
[2019-06-11T18:26:28.015Z] "- - -" 0 - "-" "-" 0 0 63 - "-" "-" "-" "-" "127.0.0.1:9080" inbound|9080|tcp|productpage.bookinfo.svc.cluster.local 127.0.0.1:48628 10.52.0.112:9080 10.52.0.100:52664 outbound_.9080_._.productpage.bookinfo.svc.cluster.local
[2019-06-11 18:26:53.877][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:81] Caught Segmentation fault, suspect faulting address 0x0
[2019-06-11 18:26:53.877][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:69] Backtrace (use tools/stack_decode.py to get line numbers):
[2019-06-11 18:26:53.877][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #0: __restore_rt [0x7f6b4dfe7390]
[2019-06-11 18:26:53.881][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #1: Envoy::Network::FilterManagerImpl::onRead() [0x914a59]
[2019-06-11 18:26:53.884][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #2: Envoy::Network::ConnectionImpl::onReadReady() [0x9113bc]
[2019-06-11 18:26:53.888][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #3: Envoy::Network::ConnectionImpl::onFileEvent() [0x910e81]
[2019-06-11 18:26:53.891][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #4: Envoy::Event::FileEventImpl::assignEvents()::$_0::__invoke() [0x90bd05]
[2019-06-11 18:26:53.894][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #5: event_process_active_single_queue [0xcb817d]
[2019-06-11 18:26:53.898][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #6: event_base_loop [0xcb6730]
[2019-06-11 18:26:53.901][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #7: Envoy::Event::DispatcherImpl::run() [0x90aaed]
[2019-06-11 18:26:53.904][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #8: Envoy::Server::WorkerImpl::threadRoutine() [0x905222]
[2019-06-11 18:26:53.907][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #9: Envoy::Thread::ThreadImplPosix::ThreadImplPosix()::$_0::__invoke() [0xe383f5]
[2019-06-11 18:26:53.908][55][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #10: start_thread [0x7f6b4dfdd6ba]
2019-06-11T18:26:53.973360Z     warn    Epoch 0 terminated with an error: signal: segmentation fault
```

**Comment by kyessenov** [comment]
Note that I cannot obtain a core dump. I scanned the proxy container and no such dump is generated.

**Comment by kramerul** [comment]
I had a short look at the problem.
The git sha version of [proxy](https://github.com/istio/proxy) is 542b4fda35046fc9dad5014dbe81f3d5969e63f7
The git sha version of [envoy](https://github.com/envoyproxy/envoy.git) is 829b905ca0fdc85233c3969247e53a62a52ac627

The segmentation fault is happening inside 

```
void FilterManagerImpl::onRead() {
  ASSERT(!upstream_filters_.empty());
  onContinueReading(nullptr, connection_);
}
```
at the statement before address 0x914a59

<pre>
Envoy::Network::FilterManagerImpl::onRead():
  9149e0:	55 	pushq	%rbp
  9149e1:	48 89 e5 	movq	%rsp, %rbp
  9149e4:	41 57 	pushq	%r15
  9149e6:	41 56 	pushq	%r14
  9149e8:	41 55 	pushq	%r13
  9149ea:	41 54 	pushq	%r12
  9149ec:	53 	pushq	%rbx
  9149ed:	50 	pushq	%rax
  9149ee:	48 8b 5f 18 	movq	24(%rdi), %rbx
  9149f2:	4c 8d 6f 18 	leaq	24(%rdi), %r13
  9149f6:	49 39 dd 	cmpq	%rbx, %r13
  9149f9:	74 6b 	je	107 <_ZN5Envoy7Network17FilterManagerImpl6onReadEv+0x86>
  9149fb:	4c 8b 37 	movq	(%rdi), %r14
  9149fe:	66 90 	nop
  914a00:	48 8b 43 10 	movq	16(%rbx), %rax
  914a04:	80 78 30 00 	cmpb	$0, 48(%rax)
  914a08:	75 13 	jne	19 <_ZN5Envoy7Network17FilterManagerImpl6onReadEv+0x3d>
  914a0a:	c6 40 30 01 	movb	$1, 48(%rax)
  914a0e:	48 8b 78 20 	movq	32(%rax), %rdi
  914a12:	48 8b 07 	movq	(%rdi), %rax
  914a15:	ff 50 18 	callq	*24(%rax)
  914a18:	83 f8 01 	cmpl	$1, %eax
  914a1b:	74 49 	je	73 <_ZN5Envoy7Network17FilterManagerImpl6onReadEv+0x86>
  914a1d:	49 8b 06 	movq	(%r14), %rax
  914a20:	4c 89 f7 	movq	%r14, %rdi
  914a23:	ff 50 10 	callq	*16(%rax)
  914a26:	49 89 c7 	movq	%rax, %r15
  914a29:	41 89 d4 	movl	%edx, %r12d
  914a2c:	48 8b 00 	movq	(%rax), %rax
  914a2f:	4c 89 ff 	movq	%r15, %rdi
  914a32:	ff 50 60 	callq	*96(%rax)
  914a35:	41 f6 c4 01 	testb	$1, %r12b
  914a39:	75 05 	jne	5 <_ZN5Envoy7Network17FilterManagerImpl6onReadEv+0x60>
  914a3b:	48 85 c0 	testq	%rax, %rax
  914a3e:	74 1e 	je	30 <_ZN5Envoy7Network17FilterManagerImpl6onReadEv+0x7e>
  914a40:	48 8b 43 10 	movq	16(%rbx), %rax
  914a44:	48 8b 78 20 	movq	32(%rax), %rdi
  914a48:	41 80 e4 01 	andb	$1, %r12b
  914a4c:	48 8b 07 	movq	(%rdi), %rax
  914a4f:	41 0f b6 d4 	movzbl	%r12b, %edx
  914a53:	4c 89 fe 	movq	%r15, %rsi
  <b>914a56:	ff 50 10 	callq	*16(%rax)</b>
  914a59:	83 f8 01 	cmpl	$1, %eax
  914a5c:	74 08 	je	8 <_ZN5Envoy7Network17FilterManagerImpl6onReadEv+0x86>
  914a5e:	48 8b 1b 	movq	(%rbx), %rbx
  914a61:	49 39 dd 	cmpq	%rbx, %r13
  914a64:	75 9a 	jne	-102 <_ZN5Envoy7Network17FilterManagerImpl6onReadEv+0x20>
  914a66:	48 83 c4 08 	addq	$8, %rsp
  914a6a:	5b 	popq	%rbx
  914a6b:	41 5c 	popq	%r12
  914a6d:	41 5d 	popq	%r13
  914a6f:	41 5e 	popq	%r14
  914a71:	41 5f 	popq	%r15
  914a73:	5d 	popq	%rbp
  914a74:	c3 	retq
  914a75:	66 66 2e 0f 1f 84 00 00 00 00 00 	nopw	%cs:(%rax,%rax)
</pre>


From this point on, unfortunately, I cannot contribute any more as I am neither an istio nor an envoy expert. Probably we have to open an issue for envoy.

**Comment by kramerul** [comment]
For me this looks like accessing an uninitialised class instance. At first sight, there are no pointers involved. Perhaps the class instance is not yet initialised or used after destruction.

**Comment by kyessenov** [comment]
Thanks @kramerul . This does look like a problem with base envoy / control plane interaction, and not an issue with istio custom filters. From my part, I've been trying unsuccessfully to reproduce it on a standalone envoy, but no luck so far.

**Comment by duderino** [comment]
@crazyxy 1.2 release blocker for you, please jump on this

CC: @PiotrSikora @jplevyak @htuch @silentdai

**Comment by kyessenov** [comment]
I left it running, so if you figure out how to get a core dump let me know. All I need is to curl some URL to trigger a seg fault.

**Comment by Multiply** [comment]
@duderino the issue I reported in #14331 is happening on 1.1.x, so a backport would be nice as well.

**Comment by silentdai** [comment]
@crazyxy 
My 2 cents: `FilterManagerImpl::onRead` is too short to fail. It's likely `FilterManagerImpl::onContinueReading` is inlined. Need symbol table to verify.

**Comment by crazyxy** [comment]
@kyessenov Could you try following way to get the core dump?

> You can enable core dumps using helm template --set global.proxy.enableCoreDump=true [other options], and once Envoy crashes, you should find it in /var/lib/istio/core.proxy.

**Comment by kyessenov** [comment]
@crazyxy given that you could not reproduce it, I'd like to avoid re-installing istio to get a core dump since that might make my bug disappear.

**Comment by kyessenov** [comment]
@PiotrSikora do you, or anyone else, know how to get a core dump on a running pod? I have sudo, I should be able to obtain whatever privileges I need to get the dump.

**Comment by kyessenov** [comment]
@crazyxy can you please scaly your istio-policy in istio-system to 4 replicas and retry? There is a theory it has something to do with that.

**Comment by Multiply** [comment]
For the issue I linked above, we only had 2 replicas.

**Comment by kyessenov** [comment]
Trace-level log:
```log
[2019-06-12 18:22:27.443][82][debug][filter] [external/envoy/source/extensions/filters/listener/original_dst/original_dst.cc:18] original_dst: New connection accepted
[2019-06-12 18:22:27.443][82][debug][filter] [external/envoy/source/extensions/filters/listener/tls_inspector/tls_inspector.cc:72] tls inspector: new connection accepted
[2019-06-12 18:22:27.444][82][trace][filter] [external/envoy/source/extensions/filters/listener/tls_inspector/tls_inspector.cc:141] tls inspector: recv: 1280
[2019-06-12 18:22:27.444][82][debug][filter] [external/envoy/source/extensions/filters/listener/tls_inspector/tls_inspector.cc:118] tls:onServerName(), requestedServerName: outbound_.9080_._.productpage.bookinfo.svc.cluster.local
[2019-06-12 18:22:27.444][82][trace][filter] [external/envoy/source/extensions/filters/listener/tls_inspector/tls_inspector.cc:162] tls inspector: done: true
[2019-06-12 18:22:27.444][82][debug][filter] [src/envoy/tcp/mixer/filter.cc:29] Called tcp filter: Filter
[2019-06-12 18:22:27.444][82][debug][filter] [src/envoy/tcp/mixer/filter.cc:39] Called tcp filter: initializeReadFilterCallbacks
[2019-06-12 18:22:27.444][82][debug][filter] [external/envoy/source/common/tcp_proxy/tcp_proxy.cc:200] [C10] new tcp proxy session
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:288] [C10] readDisable: enabled=true disable=true
[2019-06-12 18:22:27.444][82][debug][filter] [src/envoy/tcp/mixer/filter.cc:132] [C10] Called tcp filter onNewConnection: remote 10.52.0.100:54424, local 10.52.1.137:9080
[2019-06-12 18:22:27.444][82][debug][filter] [external/envoy/source/common/tcp_proxy/tcp_proxy.cc:343] [C10] Creating connection to cluster inbound|9080|tcp|productpage.bookinfo.svc.cluster.local
[2019-06-12 18:22:27.444][82][debug][pool] [external/envoy/source/common/tcp/conn_pool.cc:80] creating a new connection
[2019-06-12 18:22:27.444][82][debug][pool] [external/envoy/source/common/tcp/conn_pool.cc:372] [C11] connecting
[2019-06-12 18:22:27.444][82][debug][connection] [external/envoy/source/common/network/connection_impl.cc:702] [C11] connecting to 127.0.0.1:9080
[2019-06-12 18:22:27.444][82][debug][connection] [external/envoy/source/common/network/connection_impl.cc:711] [C11] connection in progress
[2019-06-12 18:22:27.444][82][debug][pool] [external/envoy/source/common/tcp/conn_pool.cc:106] queueing request due to no available connections
[2019-06-12 18:22:27.444][82][debug][main] [external/envoy/source/server/connection_handler_impl.cc:257] [C10] new connection
[2019-06-12 18:22:27.444][82][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:144] item added to deferred deletion list (size=1)
[2019-06-12 18:22:27.444][82][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:64] clearing deferred deletion list (size=1)
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:456] [C10] socket event: 2
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:541] [C10] write ready
[2019-06-12 18:22:27.444][82][debug][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:168] [C10] handshake error: 2
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:456] [C11] socket event: 2
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:541] [C11] write ready
[2019-06-12 18:22:27.444][82][debug][connection] [external/envoy/source/common/network/connection_impl.cc:550] [C11] connected
[2019-06-12 18:22:27.444][82][debug][pool] [external/envoy/source/common/tcp/conn_pool.cc:293] [C11] assigning connection
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:288] [C10] readDisable: enabled=false disable=false
[2019-06-12 18:22:27.444][82][debug][filter] [external/envoy/source/common/tcp_proxy/tcp_proxy.cc:542] TCP:onUpstreamEvent(), requestedServerName: outbound_.9080_._.productpage.bookinfo.svc.cluster.local
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:456] [C10] socket event: 2
[2019-06-12 18:22:27.444][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:541] [C10] write ready
[2019-06-12 18:22:27.444][82][debug][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:168] [C10] handshake error: 2
[2019-06-12 18:22:27.445][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:456] [C10] socket event: 3
[2019-06-12 18:22:27.445][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:541] [C10] write ready
[2019-06-12 18:22:27.445][82][debug][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:157] [C10] handshake complete
[2019-06-12 18:22:27.445][82][debug][filter] [src/envoy/tcp/mixer/filter.cc:171] Called tcp filter onEvent: 2 upstream 127.0.0.1:9080
[2019-06-12 18:22:27.445][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:494] [C10] read ready
[2019-06-12 18:22:27.445][82][trace][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:73] [C10] ssl read returns: 1193
[2019-06-12 18:22:27.445][82][trace][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:73] [C10] ssl read returns: -1
[2019-06-12 18:22:27.445][82][trace][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:148] [C10] ssl read 1193 bytes into 1 slices
[2019-06-12 18:22:27.445][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:288] [C10] readDisable: enabled=true disable=true
[2019-06-12 18:22:27.446][82][debug][filter] [src/istio/control/client_context_base.cc:133] Check attributes: attributes {
  key: "connection.id"
  value {
    string_value: "87725317-d334-48b5-b134-b44322b7f17b-10"
  }
}
attributes {
  key: "connection.mtls"
  value {
    bool_value: true
  }
}
att
[2019-06-12 18:22:27.446][82][debug][filter] [src/istio/mixerclient/client_impl.cc:79] Policy cache hit=true, status=PERMISSION_DENIED:whitelistip.bookinfo:10.52.0.100 is not whitelisted
[2019-06-12 18:22:27.446][82][debug][filter] [src/envoy/tcp/mixer/filter.cc:140] Called tcp filter completeCheck: PERMISSION_DENIED:whitelistip.bookinfo:10.52.0.100 is not whitelisted
[2019-06-12 18:22:27.446][82][trace][connection] [external/envoy/source/common/network/connection_impl.cc:288] [C10] readDisable: enabled=false disable=false
[2019-06-12 18:22:27.446][82][debug][connection] [external/envoy/source/common/network/connection_impl.cc:101] [C10] closing data_to_write=0 type=1
[2019-06-12 18:22:27.446][82][debug][connection] [external/envoy/source/common/network/connection_impl.cc:188] [C10] closing socket: 1
[2019-06-12 18:22:27.446][82][debug][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:270] [C10] SSL shutdown: rc=0
[2019-06-12 18:22:27.446][82][debug][connection] [external/envoy/source/extensions/transport_sockets/tls/ssl_socket.cc:201] [C10]
[2019-06-12 18:22:27.446][82][debug][filter] [src/envoy/tcp/mixer/filter.cc:171] Called tcp filter onEvent: 1 upstream 127.0.0.1:9080
[2019-06-12 18:22:27.446][82][debug][filter] [src/istio/control/client_context_base.cc:140] Report attributes: attributes {
  key: "check.cache_hit"
  value {
    bool_value: true
  }
}
attributes {
  key: "check.error_code"
  value {
    int64_value: 7
  }
}
attributes {
  key: "check.error_message
[2019-06-12 18:22:27.446][82][debug][connection] [external/envoy/source/common/network/connection_impl.cc:101] [C11] closing data_to_write=0 type=1
[2019-06-12 18:22:27.446][82][debug][connection] [external/envoy/source/common/network/connection_impl.cc:188] [C11] closing socket: 1
[2019-06-12 18:22:27.446][82][debug][pool] [external/envoy/source/common/tcp/conn_pool.cc:121] [C11] client disconnected
[2019-06-12 18:22:27.446][82][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:144] item added to deferred deletion list (size=1)
[2019-06-12 18:22:27.446][82][debug][main] [external/envoy/source/server/connection_handler_impl.cc:68] [C10] adding to cleanup list
[2019-06-12 18:22:27.446][82][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:144] item added to deferred deletion list (size=2)
[2019-06-12 18:22:27.446][82][debug][filter] [src/envoy/tcp/mixer/filter.cc:101] [C10] Called tcp filter onRead bytes: 1193
[2019-06-12 18:22:27.446][82][trace][filter] [external/envoy/source/common/tcp_proxy/tcp_proxy.cc:459] [C10] downstream connection received 1193 bytes, end_stream=false
[2019-06-12 18:22:27.446][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:81] Caught Segmentation fault, suspect faulting address 0x0
[2019-06-12 18:22:27.446][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:69] Backtrace (use tools/stack_decode.py to get line numbers):
[2019-06-12 18:22:27.447][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #0: __restore_rt [0x7f393bd75390]
[2019-06-12 18:22:27.451][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #1: Envoy::Network::FilterManagerImpl::onRead() [0x914a59]
[2019-06-12 18:22:27.455][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #2: Envoy::Network::ConnectionImpl::onReadReady() [0x9113bc]
[2019-06-12 18:22:27.459][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #3: Envoy::Network::ConnectionImpl::onFileEvent() [0x910e81]
[2019-06-12 18:22:27.463][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #4: Envoy::Event::FileEventImpl::assignEvents()::$_0::__invoke() [0x90bd05]
[2019-06-12 18:22:27.467][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #5: event_process_active_single_queue [0xcb817d]
[2019-06-12 18:22:27.471][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #6: event_base_loop [0xcb6730]
[2019-06-12 18:22:27.475][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #7: Envoy::Event::DispatcherImpl::run() [0x90aaed]
[2019-06-12 18:22:27.479][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #8: Envoy::Server::WorkerImpl::threadRoutine() [0x905222]
[2019-06-12 18:22:27.482][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #9: Envoy::Thread::ThreadImplPosix::ThreadImplPosix()::$_0::__invoke() [0xe383f5]
[2019-06-12 18:22:27.535][82][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #10: start_thread [0x7f393bd6b6ba]
2019-06-12T18:22:27.542064Z     warn    Epoch 0 terminated with an error: signal: segmentation fault
```

**Comment by crazyxy** [comment]
@kyessenov I cannot reproduce after upscaling to 4 replicas.

**Comment by howardjohn** [comment]
I reproduced this with 1 replica. Make sure you install with polciy enabled (it is off by default). I did it with demo install

**Comment by htuch** [comment]
This looks like we're still getting an `onRead` for connection C10 after it's been added to the deferred deletion list and after a remote close. That's pretty weird, seems a legit bug.

**Comment by crazyxy** [comment]
When receiving read events, filter manager will iterator all filters and executes onData for each filter. When a filter executes onData, it could close the connection (e.g. the mixer filter). For following filters in the loop, they could hit segment fault due to the connection is closed. I cannot pinpoint the exact code which causes the segment fault because of the lack of logs. I will try to reproduce again.

**Comment by crazyxy** [comment]
Upstream pr https://github.com/envoyproxy/envoy/pull/7256

**Comment by duderino** [comment]
@crazyxy can you make sure this gets into both the 1.1 and 1.2 branches?

**Comment by crazyxy** [comment]
https://github.com/istio/proxy/pull/2277
https://github.com/istio/proxy/pull/2278

**Comment by kyessenov** [comment]
Can you please the docker container with the fix? I can validate it locally I think by deploying a secondary `productpage` pod.

**Comment by duderino** [comment]
This made it into 1.1.9 which will be released on Monday 6/17.

It also will make it into 1.2.0 which most likely will be released on Tuesday 6/18

**Comment by duderino** [comment]
@kyessenov is this what you are looking for?  https://storage.googleapis.com/istio-build/proxy/envoy-debug-ae671c53f644061ab585b2c2e129dcbc46e918c0.tar.gz

**Comment by kyessenov** [comment]
This is a binary, which apparently doesn't run well in proxyv2 container. Is there istio PR that brings the new proxy in?

**Comment by kyessenov** [comment]
I tried RC.2 proxy in productpage and that resolved the seg fault, so this should be fixed in 1.2.
